client/src/www/app/outline_server_repository/server.ts

```diff
@@ -137,3 +148,42 @@ export class OutlineServer implements Server {
     return cipher !== undefined && OutlineServer.SUPPORTED_CIPHERS.includes(cipher);
   }
 }
+
+
+export interface ShadowsocksSessionConfig {
+  host?: string;
+  port?: number;
+  password?: string;
+  method?: string;
+  prefix?: string;
+}
+
+export const enum TunnelStatus {
+  CONNECTED,
+  DISCONNECTED,
+  RECONNECTING,
+}
+
+export type TunnelFactory = (id: string) => Tunnel;
+
+// Represents a VPN tunnel to a Shadowsocks proxy server. Implementations provide native tunneling
+// functionality through cordova.plugins.oultine.Tunnel and ElectronOutlineTunnel.
+export interface Tunnel {
```

Comment:
I feel this can be put in a dedicated `tunnel.ts`.

Reply:
It's been hard to track the server code. Putting it in another place makes it harder. This is a server implementation detail, so here makes sense.